packages/create-toolpad-app/src/generateProject.ts

```diff
@@ -1,363 +1,171 @@
-import path from 'path';
-import { PackageJson } from './packageType';
+// Common files for all apps
+import theme from './templates/theme';
+import eslintConfig from './templates/eslintConfig';
+import nextConfig from './templates/nextConfig';
+import nextTypes from './templates/nextTypes';
+import tsConfig from './templates/tsConfig';
+import readme from './templates/readme';
+import gitignore from './templates/gitignore';
+import ordersPage from './templates/ordersPage';
+
+// App router specific files
+import packageJsonApp from './templates/packageJson';
+import rootLayout from './templates/rootLayout';
+import rootPage from './templates/rootPage';
+import NavigateButton from './templates/navigateButton';
+import dashboardLayout from './templates/dashboardLayout';
+import dashboardPage from './templates/dashboardPage';
+
+// Pages router specific files
+import packageJsonPages from './templates/nextjs-pages/packageJson';
+import indexPage from './templates/nextjs-pages/indexPage';
+
+import app from './templates/nextjs-pages/app';
+import document from './templates/nextjs-pages/document';
+
+// Auth specific files for all apps
+import credentialsProvider from './templates/auth/credentialsProvider';
+import callbacks from './templates/auth/callbacks';
+import providerImport from './templates/auth/providerImport';
+import oAuthProvider from './templates/auth/oAuthProvider';
+import providerSetup from './templates/auth/providerSetup';
+import providerMap from './templates/auth/providerMap';
+import authImport from './templates/auth/import';
```

Comment:
IMO this goes way too far. Totally incomprehensible to figure out what's going on in an output file without jumping to three different source files.